File: src/programmers/level1/숫자문자열과영단어.java
package programmers.level1;

public class 숫자문자열과영단어 {
    public static void main(String[] args) {
        Solution11 solution11 = new Solution11();
        System.out.println(solution11.solution("2three45sixseven"));
    }
}

class Solution11 {

    String[] words = {"zero", "one", "two", "three", "four", "five",
            "six", "seven", "eight", "nine"};

    public int solution(String s) {
        StringBuilder answer = new StringBuilder();
        for(int i=0; i<s.length(); i++) {
            char ch = s.charAt(i);
            if(Character.isDigit(ch)){
                answer.append(ch);
            }
            else {
                for(int j=0; j<words.length; j++) {
                    String word = words[j];
                    if(s.length() - i >= word.length() && s.startsWith(word, i)) {
                        answer.append(j);
                        i += word.length()-1;
                        break;
                    }
                }
            }
        }
        return Integer.parseInt(answer.toString());
    }
}

File: src/programmers/level1/숫자문자열과영단어.kt
package programmers.level1

fun solution(s: String): Int {
    val stringToDigit = hashMapOf("zero" to 0, "one" to 1, "two" to 2,
                                  "three" to 3, "four" to 4, "five" to 5,
                                  "six" to 6, "seven" to 7, "eight" to 8, "nine" to 9)

    val stringBuilder = StringBuilder()
    var word = ""
    for (i in s.indices) {
        if (s[i].isDigit()) {
            stringBuilder.append(s[i])
        } else {
            word += s[i]
            if (stringToDigit.contains(word)) {
                stringBuilder.append(stringToDigit[word])
                word = ""
            }
        }
    }

    return stringBuilder.toString().toInt()
}